operator/test/e2e/e2e_pente_parallel_test.go

```diff
@@ -0,0 +1,280 @@
+			type ercBalanceOf struct {
+				Param0 *tktypes.HexUint256 `json:"0"`
+			}
+			var result ercBalanceOf
+			err := rpc[node].ForABI(ctx, erc20PrivateABI).
+				Private().
+				Domain("pente").
+				To(penteContract).
+				Function("balanceOf").
+				Inputs(&penteInvokeParams{
+					Group: penteGroupStars,
+					To:    *erc20StarsAddr,
+					Inputs: map[string]any{
+						"account": addr.String(),
+					},
+				}).
+				Outputs(&result).
+				From(fmt.Sprintf("%s@%s", identity, node)).
+				Call()
+			Expect(err).To(BeNil())
+			Expect(result).ToNot(BeNil())
+			return result.Param0
+		}
+
+		users := [][]string{
+			{"tara", "node1"},
+			{"hoshi", "node2"},
+			{"seren", "node3"},
+		}
+
+		It("mints some ERC-20 inside the the privacy group", func() {
+
+			for _, user := range users {
+
+				invoke := rpc["node1"].ForABI(ctx, erc20PrivateABI).
+					Private().
+					Domain("pente").
+					To(penteContract).
+					Function("mint").
+					Inputs(&penteInvokeParams{
+						Group: penteGroupStars,
+						To:    *erc20StarsAddr,
+						Inputs: map[string]any{
+							"to":     getEthAddress(user[0], user[1]),
+							"amount": with18Decimals(1000),
+						},
+					}).
+					From("tara@node1"). // operator
+					Send().
+					Wait(5 * time.Second)
+				testLog("SimpleERC20 mint transaction %s", invoke.ID())
+				Expect(invoke.Error()).To(BeNil())
+			}
+
+		})
+
+		It("check ERC-20 balance of each", func() {
+
+			for _, user := range users {
+				userBalance := getERC20Balance(user[0], user[1])
+				testLog("SimpleERC20 balance after mint to %s@%s: %s", user[0], user[1], userBalance.Int())
+				Expect(userBalance.String()).To(Equal(with18Decimals(1000).String()))
+			}
+
+		})
+
+		It("does a bunch of parallel transfers", func() {
+
+			results := make(chan error)
+			for _iUser, _user := range users {
+				iUser := _iUser
+				user := _user
+				go func() {
+					var err error
+					defer func() {
+						results <- err
+					}()
+
+					const count = 10
+					toUser := users[(iUser+1)%len(users)]
+					for i := 0; i < count && err == nil; i++ {
+						bigAmount, _ := rand.Int(rand.Reader, big.NewInt(9))
+						amount := bigAmount.Int64() + 1
+						invoke := rpc[user[1]].ForABI(ctx, erc20PrivateABI).
+							Private().
+							Domain("pente").
+							To(penteContract).
+							Function("transfer").
+							Inputs(&penteInvokeParams{
+								Group: penteGroupStars,
+								To:    *erc20StarsAddr,
+								Inputs: map[string]any{
+									"to":    getEthAddress(toUser[0], toUser[1]),
+									"value": with18Decimals(amount),
+								},
+							}).
+							From(fmt.Sprintf("%s@%s", user[0], user[1])).
+							Send().
+							Wait(10 * time.Second)
+						testLog("[%d]:%.3d/%.3d SimpleERC20 mint %d from %s@%s to %s@%s transaction %s",
+							iUser, i, count, amount, user[0], user[1], toUser[0], toUser[1], invoke.ID())
+						err = invoke.Error()
+					}
+				}()
+			}
+			for i := 0; i < len(users); i++ {
+				Expect(<-results).To(BeNil())
```

Comment:
If I am understanding the SDK and test framework correctly, then this test waits for the receipt of one transaction in the loop (for a given user) before sending the next transaction.  It is great that we are testing multiple loops, for different users in parallel but I think we also need a test that does not wait for receipts and just sends a stream of transactions for each user.

I think we should also check the effect of the transactions by doing a `ptx_call` just to prove that there wasn't any weird data corruption or overwriting states etc..   

In my manual testing, I was using a simple storage contract with an increment function.  I could call the `get` function at the end and if the total was what I expected, then I was confident that every transaction used the correct input state and applied the correct logic ( `n++` ).

Wondering if it is worth adding a similar thing here.

Reply:
This test deliberately:
- Runs thread thread concurrently
- Runs one transaction per thread sequentially

Sorry I missed the `getBalance` at the end to total up, just didn't get to it last night.